ai_ta_backend/update_materials.py

```diff
@@ -0,0 +1,91 @@
+import os
+import shutil
+import boto3
+from ai_ta_backend.vector_database import Ingest
+import hashlib
+from ai_ta_backend.aws import upload_data_files_to_s3
+from ai_ta_backend.vector_database import Ingest
+
+
+def generate_checksum(file_path):
+    md5_hash = hashlib.md5()
+    with open(file_path, "rb") as file:
+        # Read and update the hash string value in blocks
+        for byte_block in iter(lambda: file.read(4096), b""):
+            md5_hash.update(byte_block)
+    return md5_hash.hexdigest()
+
+def update_files(source_path: str, course_name: str):
+    """
+    Compares and updates files in S3 and QDRANT
+    Args:
+        source_path: path to the directory containing the files to be uploaded
+        course_name: name of the course whose files need to be updated
+    To-do:
+        1. Get S3 paths of files for given course_name
+        2. Compute checksums of every file in source_path folder
+        3. Compare checksums with S3 files - if different, upload to S3 and ingest into QDRANT
+    """
+    print("In update_files")
+
+    ingester = Ingest()
+    # Get S3 paths of files for given course_name
+    s3_files = ingester.getAll(course_name)   
+    print("s3 files: ", s3_files)
+    
+
+    # Access checksum of s3 files
+    s3_client = boto3.client('s3', aws_access_key_id=os.getenv('AWS_ACCESS_KEY_ID'),

```

Comment:
Hmm, this is really expensive to download all files from S3. 

In the end, it's the exact same as using the `Document Contexts` from Supabase SQL. I think we have to re-implement this with Supabase instead of S3. It'll be hundreds of times faster and cheaper and probably conceptually simpler too.